### PR TITLE
Support getting SSH keys from memory, pt. 2

### PR DESCRIPTION
**Pull request by mgorny**
This is PR #2989 with small fixes. The API is consistent now, and CMake checks for the function availability.

What remains to do are tests. However, I would like to get some more advice here. Right now, I see that two files are declaring SSH key auth callback:
* online/clone.c
* online/push.c

For clone.c, the callback is hidden a few functions deep. So if I were to introduce an alternative variant using memory credentials, I'd have to pass an additional boolean through all those functions.

For push.c, I don't see the callback being used anywhere…

In either case, we won't be able to test it properly on Travis until libssh2 1.6.0 is released and available there.

### CODE REVIEW

**Comment by carlosmn** [comment]
I don't know what you mean by the callback being hidden. The tests that check for this kind of thing, e.g. `test_online_clone__ssh_auth_methods`, only need to set the cred callback function to their own which creates the credentials you needed for the test.

**Comment by mgorny** [comment]
Ah, ok. I was looking for the test that actually used SSH key but I guess I can create my own.

**Comment by mgorny** [comment]
Done and tested locally! It reads the key files supplied in the env and passes them to `git_cred_ssh_key_memory_new()`. The test is skipped when in-memory keys are not supported.

File: tests/online/clone.c [hunk 1]
```diff
@@ -519,6 +519,74 @@ void test_online_clone__ssh_cert(void)
...
+	const char *remote_user = cl_getenv("GITTEST_REMOTE_USER");
+	const char *privkey = cl_getenv("GITTEST_REMOTE_SSH_KEY");
+
+#ifndef GIT_SSH_MEMORY_CREDENTIALS
+	clar__skip();
+#endif
+	if (!remote_url || !remote_user || !privkey || strncmp(remote_url, "ssh://", 5) != 0)
+		clar__skip();
+
+	g_options.remote_callbacks.credentials = ssh_memory_cred_cb;
```
Comment (carlosmn) [inline tests/online/clone.c]: This should be more like `g_options.fetch_options.callbacks.credentials`. It looks like the remote changes got merged after this branched off, so rebasing on top of master would be good to get the tests to use the new API.

**Comment by mgorny** [comment]
Rebased and updated now. Are there any more concerns?

**Comment by carlosmn** [comment]
:+1: from me, @ethomson anything to add?

**Pull request merged**